c7n/resources/mq.py

```diff
@@ -82,6 +83,57 @@ def get_dimensions(self, resource):
         return [{'Name': self.model.dimension,
                  'Value': "{}-1".format(resource['BrokerName'])}]
 
+@MessageBroker.filter_registry.register('vpc')
+class VpcFilter(VpcFilter):
+
+    sg_vpc_dict = {}
+
+    def get_related_ids(self, resources):
+        client = local_session(self.manager.session_factory).client('ec2')
+        related_ids = []
+        for r in resources:
+            sg_ids = r.get('SecurityGroups', [])
+            vpc_id = self.sg_vpc_dict.get(sg_ids[0])
```

Comment:
Will this work if the `sg_id` list is empty?

Reply:
It wouldn't work if sg_id was empty, but security group is a required field when making Message Brokers through AWS MQ, so I don't believe it's possible for it to be empty.